docs/Middleware-Guide.md

```diff
@@ -0,0 +1,82 @@
+
+The following figure illustrates the above flow in which the operator calls on a `deregister(..)` function in a sample Registry contract.
+![Operator deregisteringn](./images/operator_deregister.png)
+
+### *Staker opting-in with a middleware*
+In order for any staker to participate in responding to the tasks from the middleware without running the middleware's off-chain software container themself, the staker can pursue delegation. The sequential flow for delegation process is as follows:
+1. The staker first has to delegate its stake to an EigenLayer operator who is opted-in to that middleware and that they trust (based on their own due diligence). The staker can accomplish this by calling the API `delegateTo(..)` in the DelegationManager contract, where the operator's address is passed as the argument.  
+2. Observe that with the staker delegating its stake to the operator, the weight associated with the operator's response to the task in the middleware it has opted-in increases. Therefore, the middleware must be notified about this. When the middleware is notified of this update, the call the `recordStakeUpdate` functionality to recognize the change in stake for that operator. On successful execution of this API call, the event `MiddlewareTimesAdded(..)` is emitted.
+
+The following figure illustrates the above flow: 
+![Staker opting-in](./images/staker_opting.png)
+
+
+### *Staker withdraws*
+In order for a staker, who has opted-in to a middleware via delegation, wants to partially withdraw some of its stake or completely stop participating, it has to undelegate its stake from the current operator. Under current design of delegation, whenever the staker partially withdraw some of its stake or completely stop participating, then it affects all the middlewares uniformly that its operator is participating in. The sequential flow is as follows:
+1. The staker first has to queue its withdrawal request with EigenLayer. The staker can place this request by calling  `queueWithdrawal(..)` in the EigenLayer's `StrategyManager.sol`. On successful call to this, the event `WithdrawalQueued(..)` is emitted.
+2. The staker then has to notify all the middlewares that its operator has opted in about its intention to partially withdraw some of its stake or completely stop participating and also record the latest time in the EigenLayer contracts until which the staker is obliged to participate with the current stake. To do this, the staker calls on the `updateStakes(..)` function in the Registry contract that calls `recordStakeUpdate(..)` in the middleware's ServiceManager contract which in turn accesses `recordStakeUpdate(..)` in the EigenLayer's Slasher contract.  On successful execution of this call, the event `MiddlewareTimesAdded(..)` is emitted.
```

Comment:
please fix this wording "The staker then has to notify all the middlewares that its operator has opted in about its intention to partially withdraw some of its stake or completely stop participating and also record the latest time in the EigenLayer contracts until which the staker is obliged to participate with the current stake"
there's something bad happening here -- feels like 2 sentences got chopped together or something. also seems inaccurate from a technical perspective.

Reply:
yeah good point, didn't catch this.  I think Im actually gonna axe this section and the Staker delegation section.  I don't think its as relevant, I will talk to Soubhik.